src/compiler/emitter.ts

```diff
@@ -2617,20 +2617,22 @@ module ts {
                 }
             }
 
-            function getBlockScopedVariableId(node: Identifier): number {
-                // return undefined for synthesized nodes
-                return !nodeIsSynthesized(node) && resolver.getBlockScopedVariableId(node);
+            function getGeneratedNameForIdentifier(node: Identifier, isValueOfShorthandPropertyAssignment: boolean): string {
+                if (!generatedBlockScopeNames || nodeIsSynthesized(node)) {

```

Comment:
current assumption is that synthesized nodes should be initially created with correct names. Also since synthesized nodes are created in the emitter - checker is unaware of them, there is not symbol for them and proper id cannot be generated